atc/worker/artifact_source.go

```diff
@@ -60,6 +61,14 @@ func (source *artifactSource) StreamTo(
 	logger lager.Logger,
 	destination ArtifactDestination,
 ) error {
+	ctx, span := tracing.StartSpan(ctx, "artifactSource.StreamTo", nil)
+	defer span.End()
+
+	_, outSpan := tracing.StartSpan(ctx, "volume.StreamOut", tracing.Attrs{
+		"origin-volume": source.volume.Handle(),
+		"origin-worker": source.volume.WorkerName(),
+	})
+	defer outSpan.End()
```

Comment:
oh, reminder: it'd be good to capture the error so that one can see from the traces when failures occur :grin:

Reply:
@cirocosta Hmm how exactly would I be able to go about doing that? Is there an example somewhere of a trace capturing when failures occur? I also have a question about why there is another span for `volume.StreamOut` created at the same time as the `artifactSource.StreamTo` span? Is the `volume.StreamOut` span supposed to be passed into the `source.volume.StreamOut` method?

Reply:
In our latest change, we used `tracing.End()` to address the case when streaming fails. 

The reason we used 2 spans here to capture the `source` volume handle and worker name as well as the `destination` volume handle and worker name.  Furthermore, it also gives us the ability to treat these as 2 separate operations ( `source to atc` and `atc to destination` ) for performance monitoring as well as error handling.